GetJobber/atlantis
# fix(components): Update toolbar UI to float below InputText

## Motivations

The InputText height grows upon focus to accommodate the toolbar. The problem with this is that it leads to lots of UI jumping around, causing a bad experience.

**The Solution**
Stop auto-expanding the textarea height and instead slide the toolbar up from the bottom of the input with absolute positioning so it doesn't affect the layout.


## Before

https://github.com/user-attachments/assets/33adb5b2-bfb3-40b4-859f-3813732d9224

### Dark mode
<img width="1202" alt="Screenshot 2025-01-07 at 9 54 00 AM" src="https://github.com/user-attachments/assets/652e658e-4589-4ec0-8df1-bc4b92a85df4" />


## After

https://github.com/user-attachments/assets/18a8ab8a-9e84-49af-bb65-d5c99c2b71f3

### Dark mode
<img width="1214" alt="Screenshot 2025-01-07 at 9 54 11 AM" src="https://github.com/user-attachments/assets/0397549b-8172-4a3f-a496-b817bd5dd228" />





## Changes



### Changed

- The `toolbar` content now floats below the `InputText` instead of auto-expanding to fit it
- The `resize` handle has moved back to the textarea input element 
    - note: this was how it worked ~7 months ago 

### Fixed

- Fixed the safari scrollbar to no longer be offset from the right side _and_ to not block the `resize` handle
- Fixed a double scrollbar appearing in some browsers
- Prevent showing the toolbar when the input is `disabled`

## Testing

To test this:

* Pull this branch down, open the Toolbar story [locally](http://localhost:6005/?path=/story/components-forms-and-inputs-inputtext-web--toolbar)
* Focus the input field, observe the toolbar appears below the input
* Blur the input field, observe the toolbar fades away
* Force scrollbars to be visible "Always" in the macOS Settings -> Appearance menu
    * Observe the input behaves as you would expect 
* Repeat in other browsers
    * Safari: Test the scrollbar, verify the `resize` handle is not blocked/still functional 

### Browserstack

I tested various browsers on windows.

#### Edge

https://github.com/user-attachments/assets/07fc466b-69df-4f66-b223-0578bf1fb5a4


#### Firefox

https://github.com/user-attachments/assets/75278ebc-762e-4fda-bdd2-2c4e1d989bc1


#### Chrome

https://github.com/user-attachments/assets/a1b17cd5-5afd-42fa-a370-7a0195350e4a



Changes can be [tested via Pre-release](https://github.com/GetJobber/atlantis/actions/workflows/trigger-qa-build.yml)
